server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClientImpl.java

```diff
@@ -106,9 +108,33 @@ public ListenableFuture<List<DataSegment>> fetchUsedSegments(String dataSource,
     );
   }
 
+  @Override
+  public ListenableFuture<List<PhysicalDatasourceMetadata>> fetchDatasourceMetadata(List<String> dataSources)
+  {
+    String path = "/druid/coordinator/v1/metadata/datasourceMetadata";
```

Comment:
it might be better for this API to be a POST API where the expected request payload is a list of datasources.

Reply:
Also, we should use a different term than datasource metadata. since that's something used by Overlord to represent checkpoints etc.

Reply:
I can use `datasourceSchema` or `datasourceSignature` instead.